apps/dgst.c

```diff
@@ -579,5 +587,11 @@ int do_fp(BIO *out, unsigned char *buf, BIO *bp, int sep, int binout,
         }
         BIO_printf(out, "\n");
     }
-    return 0;
+
+    ret = 0;
+ end:
+    if (NULL != sigbuf) {
```

Comment:
Normally, we tend to do this the other way around, i.e. `sigbuf != NULL`

Reply:
Such practice could avoid misspelling:
e.g:

```c
int a = 42;

if (a = 3) 
   abort(); // you would always abort

if (3 == a) // you would got compilation error here
   abort();
```

Reply:
on the other hand its more difficult to read (at least for lt gt) and afaik most modern compilers would emit a warning telling you to put additional braces around the assignment to indicate that is really what you want to do.

Reply:
Thank you, I've updated this line